src/ctia/lib/time.clj

```diff
@@ -105,3 +105,10 @@
   (let [inf-range (periodic-seq start step)
         below-end? (fn [t] (time/within? (time/interval start end) t))]
     (take-while below-end? inf-range)))
+
+(defn date-str->valid-time [date-str units offset]
+  "Create a ctim.schemas.common/ValidTime from a date str and an offset"
+  (let [formatter (time-format/formatters :date-time-no-ms)
+        start (time-format/parse date-str)]
+    {:start_time (time-coerce/to-date start)
+     :end_time (time-coerce/to-date (plus-n units start offset))}))
```

Comment:
may I suggest to put the units in third position and make it optional with a default value?

~~~clojure
(defn date-str->valid-time
  ([date-str offset]
    (date-str->valid-time date-str offset :days))
 ([date-str offset units]
   "Create a ctim.schemas.common/ValidTime from a date str and an offset"
   (let [formatter (time-format/formatters :date-time-no-ms)
         start (time-format/parse date-str)]
     {:start_time (time-coerce/to-date start)
      :end_time (time-coerce/to-date (plus-n units start offset))})))
~~~